.github/workflows/release.yaml

```diff
@@ -248,22 +250,28 @@ jobs:
         env:
           GNUPGHOME: ${{ env.GPG_HOMEDIR }}
 
-      # For non-rc releases, create and push the release branch
+      # For non-rc releases that starts from main, create the release branch as it suggests that 
+      # this is not a patch release
       - name: Create and push dot release branch to public repository
-        if: env.IS_RC == 'false'
+        if: env.IS_RC == 'false' && env.BRANCH_IS_MAIN == 'true'
         run: |
           git lfs fetch --all
           git checkout -b "${{ env.RELEASE_BRANCH_NAME }}"
-          git push
-          git checkout "${{ github.ref_name }}"
 
-      # Push the tag related to the current version
-      - name: Push tag to public repository
+      # Create the tag related to the current version
+      - name: Create tag
         run: |
           git fetch --tags --force
           git tag -s -a -m "${{ env.GIT_TAG }} release" "${{ env.GIT_TAG }}"
-          git push origin "refs/tags/${{ env.GIT_TAG }}"
 
+      # Push changes to ref
+      - name: Push changes
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.BOT_TOKEN }}
+          branch: ${{ github.ref }}
+          tags: true
+  
```

Comment:
yes I want to always push the tag + push the new branch if it's a dot release (not a patch). But since I don't commit anything else I think it's fine, I'll add a comment in a following PR thanks for pointing that out